v3/mqtt_mode.go

```diff
@@ -60,49 +62,80 @@ type (
 		dc            *DeviceContext
 		subscriptions map[string]mqttMsgHandler
 
-		requestTimeout time.Duration
+		receiveQueueSize uint16
+		sendQueueSize    uint16
+		responseTimeout  time.Duration
 
 		// For handling our subscriptions. Output
 		command chan<- *DeviceCommand
 		kvSync  chan<- *KeyValueSync
 
-		// For receiving data from the API. Input (but we create and manage
-		// them)
-		SubRecvCh  chan MqttSubData
-		QueueAckCh chan MqttQueueResult
-		PingAckCh  chan bool
+		// For receiving data from the API. Input
+		SubRecvCh  <-chan MqttSubData
+		QueueAckCh <-chan MqttResponse
+		PingAckCh  <-chan MqttResponse
 	}
 )
 
+type (
+	ModeMqttOptDuration    time.Duration
+	ModeMqttOptQueueLength int16
+
+	ModeMqttDelegateOpt interface {
+		apply(d *ModeMqttDelegate)
+	}
+)
+
```

Comment:
I think, it better to exist compile-time type checking.

https://medium.com/@matryer/golang-tip-compile-time-checks-to-ensure-your-type-satisfies-an-interface-c167afed3aae

```suggestion
var _ MqttDelegate = (*ModeMqttDelegate)(nil)
```

Reply:
Good idea!